XiEditor/StyleMap.swift

```diff
@@ -59,17 +103,28 @@ func utf8_offset_to_utf16(_ s: String, _ ix: Int) -> Int {
 
 /// A store of text styles, indexable by id.
 class StyleMap {
+    private var font: NSFont
     private var map: [Style?] = []
 
+    init(font: NSFont) {
+        self.font = font
+    }
+
     func defStyle(json: [String: AnyObject]) {
         guard let styleID = json["id"] as? Int else { return }
-        let fgColor = json["fg_color"] as? UInt32 ?? 0xFF000000
-        let bgColor = json["bg_color"] as? UInt32 ?? 0
-        let weight = json["weight"] as? UInt16 ?? 400
+
+        let fgColor = colorFromArgb(json["fg_color"] as? UInt32 ?? 0xFF000000)
+        let bgColor = colorFromArgb(json["bg_color"] as? UInt32 ?? 0)
         let underline = json["underline"] as? Bool ?? false
         let italic = json["italic"] as? Bool ?? false
+        var weight = json["weight"] as? Int
+        if let w = weight {
+            // convert to NSFont weight: (100-500 -> 2-6 (5 normal weight), 600-800 -> 8-10, 900 -> 12
```

Comment:
i'd be tempted to include a link to the discussion issue inline, for future archaeologists